### PR TITLE
KAADO-44: Create edit card page

### PR DESCRIPTION
**Pull request by tsoibet**
- Create edit card page
- Fix updateCard bug

### CODE REVIEW

File: src/app/cards/edit/[id]/Form.tsx [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+    });
+    const [loading, setLoading] = useState(false);
+    const [errorMsg, setErrorMsg] = useState('');
+
+    const router = useRouter();
+
+    function handleImgChange(event: ChangeEvent<HTMLInputElement>) {
+        const { name } = event.target;
+        const file = event.target.files && event.target.files[0];
+        console.log(file);
```
Comment (nasudon) [inline src/app/cards/edit/[id]/Form.tsx]: Do we need to log this?

File: src/app/cards/edit/[id]/Form.tsx [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+                }
+                setCard({ ...card, [name]: resizedImgStr ?? '' });
+            }
+        );
+    }
+
+    async function saveBtnClickHandler(event: React.FormEvent) {
+        event.preventDefault();
+        setLoading(true);
+        console.log('clicked');
```
Comment (nasudon) [inline src/app/cards/edit/[id]/Form.tsx]: Do we need to log this?

File: src/app/cards/edit/[id]/Form.tsx [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+                    <p className="text-xs">FRONT</p>
+                    <input
+                        id="image_front"
+                        name="image_front"
+                        type="file"
+                        accept="image/*"
+                        onChange={(event) => handleImgChange(event)}
+                        disabled={loading}
+                        className="h-0"
+                    />
```
Comment (nasudon) [inline src/app/cards/edit/[id]/Form.tsx:142]: I think we should "hide" the default input

File: src/app/cards/edit/[id]/Form.tsx [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+                    <p className="text-xs">BACK</p>
+                    <input
+                        id="image_back"
+                        name="image_back"
+                        type="file"
+                        accept="image/*"
+                        onChange={(event) => handleImgChange(event)}
+                        disabled={loading}
+                        className="h-0"
+                    />
```
Comment (nasudon) [inline src/app/cards/edit/[id]/Form.tsx:166]: Same here. I think we should "hide" the default input

File: src/app/cards/edit/[id]/Form.tsx [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+                        }`}
+                    >
+                        <img
+                            src={card.image_back || 'https://placehold.co/600x400?text=NoImage'}
+                            alt="Image Preview for Back side"
+                            className="w-full h-full object-cover object-center"
+                        />
+                    </label>
+                </div>
+
```
Comment (nasudon) [inline src/app/cards/edit/[id]/Form.tsx]: [NIT] Unused new line

File: src/app/cards/edit/[id]/Form.tsx [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+                <div className="w-full flex flex-col gap-0.5">
+                    <label htmlFor="name" className="text-xs self-start">
+                        CARD NAME
+                    </label>
+                    <input
+                        id="name"
+                        type="text"
+                        value={card.name}
+                        onChange={(event) => setCard({ ...card, name: event.target.value })}
```
Comment (nasudon) [inline src/app/cards/edit/[id]/Form.tsx]: [NIT] The `(event) => setCard({ ...card, name: event.target.value })` repeated many times in this file. Shall we move it out as a function?

File: src/app/cards/edit/[id]/page.tsx [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+        const { card: updatedCard, error } = await updateCard(params.id, {
+            image_front,
+            image_back,
+            name,
+            type_id,
+            number,
+            note,
+        });
+        if (updatedCard) {
+            revalidatePath('/');
```
Comment (nasudon) [inline src/app/cards/edit/[id]/page.tsx:36]: Shall we revalidate the `/cards/detail/${params.id}` as well?

**Review by nasudon — CHANGES_REQUESTED** [review]
Please check my comments. Some of them are just nitpick and please focus on the UI part

File: src/app/cards/edit/[id]/Form.tsx (hunk 1 above)
Comment (tsoibet) [reply src/app/cards/edit/[id]/Form.tsx]: Fixed

File: src/app/cards/edit/[id]/Form.tsx (hunk 3 above)
Comment (tsoibet) [reply src/app/cards/edit/[id]/Form.tsx:142]: Fixed

File: src/app/cards/edit/[id]/Form.tsx (hunk 4 above)
Comment (tsoibet) [reply src/app/cards/edit/[id]/Form.tsx:166]: Fixed

File: src/app/cards/edit/[id]/Form.tsx (hunk 5 above)
Comment (tsoibet) [reply src/app/cards/edit/[id]/Form.tsx]: Fixed

File: src/app/cards/edit/[id]/page.tsx (hunk 7 above)
Comment (tsoibet) [reply src/app/cards/edit/[id]/page.tsx:36]: Fixed.
Added `/cards/${params.id}` too.

File: src/app/cards/edit/[id]/Form.tsx (hunk 6 above)
Comment (tsoibet) [reply src/app/cards/edit/[id]/Form.tsx]: Fixed

File: src/app/cards/edit/[id]/Form.tsx (hunk 2 above)
Comment (tsoibet) [reply src/app/cards/edit/[id]/Form.tsx]: Fixed

**Review by nasudon — APPROVED** [review]
LGTM

**Pull request merged**